inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasUimaParser.java

```diff
@@ -232,78 +213,162 @@ else if (annotation instanceof Sentence) {
+                                    
+                                    // Indexing UI annotation without type and layer for generic search
+                                    indexedStr = IndexingConstants.kbEntity + MtasToken.DELIMITER + kbValue;
+                                    log.debug("Indexed String without type and label for : {}", indexedStr);
+                                    MtasToken mtasAnnotationKBEntity = new MtasTokenString(
+                                            mtasId++, indexedStr, beginToken);
+                                    mtasAnnotationKBEntity.setOffset(annotation.getBegin(),
+                                            annotation.getEnd());
+                                    mtasAnnotationKBEntity.addPositionRange(beginToken,
+                                            endToken);
+                                    tokenCollection.add(mtasAnnotationKBEntity);
+                                }
                             }
                         }
                     }
                 }
             }
-        }
-        catch (Exception e) {
+        } catch (Exception e) {
             log.error("Unable to index document", e);
         }
         return tokenCollection;
     }
 
-    @Override
-    public String printConfig()
+    /**
+     * Replaces space with underscore n a {@code String}
+     * @param {@code String} uiName
+     * @return {@code String}
+     */
+    public String getIndexedName(String uiName)
     {
+        String indexedName = uiName.replace(" ", "_");
+        return indexedName;
+    }
+
+    @Override
+    public String printConfig() {
+        return null;
+    }
+    /**
+     * 
+     * Takes in {@code IRI} for identifier as {@code String} and returns the label String 
+     * Eg:- InputParameter :- {@code http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#RoseDAnjou} 
+     * Returned :- {@link KBConcept} + {@link MtasToken.DELIMITER} + RoseDAnjou
+     * @param String iri 
+     * @return String {@link KBObject}+{@link MtasToken.DELIMITER}
+     *          +{@link KBObject.get().getUiLabel()}
+     */
+    public String getUILabel(String aIRI) {
+        StringBuilder labelStr = new StringBuilder();
+        System.out.println();
+        Optional<KBObject> kbObject = KBUtility.readKBIdentifier(kbService,project, aIRI);
+        if (kbObject.isPresent()) {
+            labelStr.append(kbObject.get().getClass().getSimpleName())
+            .append(MtasToken.DELIMITER).append(kbObject.get().getUiLabel());
+        } else {
+            return labelStr.toString();
+        }
+        return labelStr.toString();
+    }
+    
+    /**
+     * Method Implementation to get MtasToken (To be done when subclass and class 
+     * semantics to be included
+     * @return
+     */
+    public MtasToken getMtasTokenKBFeature() {
         return null;
     }
 
-    class SimpleAnnotationLayer
-    {
+    class SimpleAnnotationLayer {
```

Comment:
Didn't you say the class was unused?

Reply:
We can remove it, but I am not sure if its written for some specific future work. @betoboullosa can we remove this?